feat: Add jwt leeway configuration option

frankdekker opened the pull request:
When a jwt token is created on one server and then used on another, there might be a slight time difference in the jwt token timestamp. However the jwt validation is by default set to `PT0S` leeway and rejects the jwt token.

The league/oauth2-server uses lcobucci/jwt for jwt creation and validation. There's an option to set the jwt leeway. See
the constructor of https://github.com/thephpleague/oauth2-server/blob/master/src/AuthorizationValidators/BearerTokenValidator.php

However there is no way to set this value via the oauth2-server-bundle. This PR allows to set this value.

chalasr commented:
Failure is not related actually. I'm going to have a look asap, help welcome.

chalasr commented:
Failure fixed, rebase needed :)

frankdekker commented:
> Failure fixed, rebase needed :)

Hey, thanks for the test fix. It seems to have improved a lot but not all the way. Looking at your PR for the fix, that's also not green yet.

chalasr commented:
Remaining failures need some work to go away, not right now :) Can you add a test case for this?

frankdekker commented:
> Remaining failures need some work to go away, not right now :) Can you add a test case for this?

Added an acceptance test. 

I had a bit of struggles with the cache in `/tests/.kernel`. Changing the TestKernel configuration was ignored by the existing cache. I added separate cache directories for when TestKernel configuration is given. Let me know if you agree with this approach, or you have another suggestion :).

chalasr reviewed (approved):
Nice 👍

chalasr commented:
@frankdekker Already suffered from this as well, we should do something to improve the situation. All good here

chalasr commented:
Thank you @frankdekker.
